app/templating/schema_context.py

```diff
@@ -14,33 +15,33 @@ def build_schema_context(metadata, aliases, answer_store, answer_ids_on_path, gr
     return {
         'exercise': _build_exercise(metadata),
         'respondent': _build_respondent(metadata),
-        'answers': _map_alias_to_answers(aliases, answer_store, answer_ids_on_path),
+        'answers': _build_answers(answer_store, answer_ids_on_path),
         'group_instance': group_instance,
     }
 
 
-def _map_alias_to_answers(aliases, answer_store, answer_ids_on_path):
-    values = {}
-    for alias, answer_info in aliases.items():
-        matching_answers = []
+def _build_answers(answer_store, answer_ids_on_path):
+    answers = {}
 
-        answers = answer_store.filter(answer_ids=[answer_info['answer_id']], limit=True)
+    for answer_id in answer_ids_on_path:
+        is_repeating_answer = _get_is_repeating_answer(answer_id)
+        answer_is_in_repeating_group = _get_answer_is_in_repeating_group(answer_id)
 
-        for answer in answers:
-            if answer['answer_id'] in answer_ids_on_path:
-                safe_answer = json_and_html_safe(answer['value'])
-                matching_answers.append(safe_answer)
+        matching_answers = answer_store.filter(
+            answer_ids=[answer_id], limit=True)
 
-        number_of_matches = len(matching_answers)
+        value = ''
```

Comment:
Would this be better served as an empty list?

Reply:
The default state for the value is an empty string. It's only if we're in a repeating group or it's a repeating answer that the value should ever be a list.